functions/function/http-request-function/src/main/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionConfiguration.java

```diff
@@ -41,17 +43,23 @@
  * @author David Turanski
  *
  **/
-@Configuration
+@Configuration(proxyBeanMethods = false)
 @EnableConfigurationProperties(HttpRequestFunctionProperties.class)
 public class HttpRequestFunctionConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean(WebClient.class)
-	public WebClient webClient(HttpRequestFunctionProperties properties) {
-		return WebClient.builder()
-				.codecs(clientCodecConfigurer ->
-						clientCodecConfigurer.defaultCodecs().maxInMemorySize(properties.getMaximumBufferSize()))
-				.build();
+	public WebClient webClient(HttpRequestFunctionProperties properties,
+			@Nullable ComponentCustomizer<WebClient.Builder> webClientComponentCustomizer) {
```

Comment:
This is a case where there is already a customizer in Boot (`org.springframework.boot.web.reactive.function.client.WebClientCustomizer`). Should we just respect that as that is what most user's would probably be familiar with?

Reply:
Will be handled in #286